### PR TITLE
ROMFS: respect kconfig for including romfs files (airframes, etc)

### PR DESCRIPTION
**Pull request by dagar**
This saves ~8.1 kB of flash on px4_fmu-v6x_default.

### CODE REVIEW

**Review by sfuhrer — DISMISSED** [review]
Good idea, I see nothing speaking against it. Helps also as a small step towards separating the airframe types.

**Review by MaEtUgR — APPROVED** [review]
Nice! Thank you for that proposal. I rebased on main and added differential rover.

File: ROMFS/px4fmu_common/init.d/CMakeLists.txt [hunk 1]
```diff
@@ -34,28 +34,79 @@
...
+if(CONFIG_MODULES_ROVER_POS_CONTROL)
+	px4_add_romfs_files(
+		rc.rover_apps
+		rc.rover_defaults
+
+		rc.boat_defaults # hack
+	)
+endif()
+
+if(CONFIG_MODULES_DIFFERENTIAL_DRIVE_CONTROL)
```
Comment (PerFrivik) [inline ROMFS/px4fmu_common/init.d/CMakeLists.txt:80]:
```suggestion
if(CONFIG_MODULES_DIFFERENTIAL_DRIVE)
```

Comment (MaEtUgR) [reply ROMFS/px4fmu_common/init.d/CMakeLists.txt:80]: Not yet. We merge this one first and adjust it in #22550 before we merge that ok?

Comment (PerFrivik) [reply ROMFS/px4fmu_common/init.d/CMakeLists.txt:80]: sounds good

**Comment by MaEtUgR** [comment]
ok, I'm testing with all the Kconfig interesting options disabled. I can only choose the balloon airframe then and dmesg complains about `SDLOG` parameters missing:
image

**Review by MaEtUgR — DISMISSED** [review]
Let's do a follow-up to clean up the balloon airframe.

**Review by MaEtUgR — APPROVED** [review]
I rebased on main and adapted the differential drive module name.
Can we merge?

**Comment by MaEtUgR** [comment]
Quickly test logger and temperature calibration being there and running. Better safe than sorry.

**Comment by PerFrivik** [comment]
Tested logger and temperature calibration :+1:, works as intended.

**Review by PerFrivik — APPROVED** [review]

**Pull request merged**